### PR TITLE
Add more LCALS kernels

### PR DESCRIPTION
**Pull request by daviditen**
Add the rest of the serial LCALS kernels.  PIC_2D and MAT_X_MAT are not getting correct checksums, so are currently disabled by default.  The rest are enabled by default.

Added the new kernels to the lists of perfkeys and kernel names to prediff for.

Once this is merged, we will need to update performance data files for the new keys.

### CODE REVIEW

File: test/studies/lcals/LCALSDataTypes.chpl [hunk 1]
```diff
@@ -322,8 +322,54 @@ module LCALSDataTypes {
     var RealArray_2D_Nx25: [0..#s_num_2D_Nx25_Real_arrays][0..#aligned_chunksize, 0..#25] real;
     var RealArray_2D_7xN: [0..#s_num_2D_7xN_Real_arrays][0..#7, 0..#aligned_chunksize] real;
     var RealArray_2D_64x64: [0..#s_num_2D_64x64_Real_arrays][0..#64, 0..#64] real;
-    var RealArray_3D_2xNx4: [0..#s_num_3D_2xNx4_Real_arrays][0..#2, 0..#aligned_chunksize, 0..#4] real;
+
+    //var RealArray_3D_2xNx4: [0..#s_num_3D_2xNx4_Real_arrays][0..#2, 0..#aligned_chunksize, 0..#4] real;
```
Comment (ronawho) [inline test/studies/lcals/LCALSDataTypes.chpl]: Did you intend to leave this comment in, or can it be removed?

File: test/studies/lcals/RunCRawLoops.chpl [hunk 2]
```diff
@@ -174,8 +257,54 @@ module RunCRawLoops {
...
+                p[ip,1] += z[j2+32];
+                i2 += e[i2+32];
+                j2 += f[j2+32];
+                h[j2,i2] += 1.0;
+              }
+            }
+            ltimer.stop();
+/*
+            writeln(ilength);
+            stdout.writef("(%{#########.#######################}, %{#########.#######################}, %{#########.#######################})\n", p[0,0], p[10,10], p[20,20]);
```
Comment (ronawho) [inline test/studies/lcals/RunCRawLoops.chpl]: Still want these comments?

**Comment by ronawho** [comment]
New kernels look good to me.

Upping the checksum tolerance has me a little concerned. Is there any easy way to leave it at .05, except for the benchmarks that aren't there yet? Do you know what the reference uses as a tolerance?

Also, I think we're getting to the point where there's too many kernels on the same graph, so we'll probably want to figure out a better way to view the data soon

File: test/studies/lcals/LCALSDataTypes.chpl (hunk 1 above)
Comment (daviditen) [inline test/studies/lcals/LCALSDataTypes.chpl]: I left this one in on purpose.  The commented out line is how it seems like that array _should_ be declared, but instead it uses the LCALS_Overlapping_Array_3D.  I can remove it if you'd prefer.

File: test/studies/lcals/RunCRawLoops.chpl (hunk 2 above)
Comment (daviditen) [inline test/studies/lcals/RunCRawLoops.chpl]: Those are a leftover from debugging.  I'll remove them.

File: test/studies/lcals/LCALSDataTypes.chpl (hunk 1 above)
Comment (ronawho) [inline test/studies/lcals/LCALSDataTypes.chpl]: I think it's ok to leave, but I wouldn't mind a brief "This is how we want to write the array declaration" comment

**Comment by daviditen** [comment]
Thank you.

I'm not worried about bumping up the tolerance a little because:

* The checksum tolerance is something I added that isn't in the reference.  The reference doesn't check the checksums against any expected values.  I added the "checksum checker" to help make sure I was getting decent answers without having to compare vs. reference output by hand.

* The README in the LCALS reference says: "Note that certain levels and types of compiler optimization will cause slight differences in checksums due to changes in operation order, for example.  Thus, the checksums may only be a qualitative indicator of correct execution."

* I've found that if there is a problem, e.g. with MAT_X_MAT or PIC_2D currently, the checksum tends to be off by a huge amount, ~10^6 or more.

You're right about the graphs getting noisy, and I haven't added the parallel versions to the graphs yet.

**Comment by ronawho** [comment]
Ok, that definitely lessons my concerns about the checksum.

Good to merge then!

**Comment by daviditen** [comment]
Thanks!

It's merged now and I used your script and instructions to update the performance .dat files.  That script rocks.

**Pull request merged**